src/pudl/transform/epacems.py

```diff
@@ -42,160 +46,243 @@ def harmonize_eia_epa_orispl(
     convert_to_utc uses the plant ID to look up timezones.
 
     Args:
-        df: A CEMS hourly dataframe for one year-month-state.
+        lf: A CEMS hourly LazyFrame for one year-month-state.
         crosswalk_df: The core_epa__assn_eia_epacamd dataframe from the database.
 
     Returns:
         The same data, with the ORISPL plant codes corrected to match the EIA plant IDs.
     """
-    # Make sure the crosswalk does not have multiple plant_id_eia values for each
-    # plant_id_epa and emissions_unit_id_epa value before reassigning IDs.
-    one_to_many = crosswalk_df.groupby(
-        ["plant_id_epa", "emissions_unit_id_epa"]
-    ).filter(
-        lambda x: x.plant_id_eia.nunique() > 1  # noqa: PD101
-    )
-    if not one_to_many.empty:
-        raise AssertionError(
-            "The core_epa__assn_eia_epacamd crosswalk has more than one plant_id_eia value per "
-            "plant_id_epa and emissions_unit_id_epa group"
-        )
-    crosswalk_df = crosswalk_df[
-        ["plant_id_eia", "plant_id_epa", "emissions_unit_id_epa"]
-    ].drop_duplicates()
-
```

Comment:
Noting for future travelers: this check got moved to `_validate_crosswalk_uniqueness` below